core/dbt/version.py

```diff
@@ -27,81 +51,175 @@ def get_latest_version(version_url: str = PYPI_VERSION_URL):
-    else:
-        latest_s = latest.to_version_string(skip_matcher=True)
 
-    version_msg = "installed version: {}\n" "   latest version: {}\n\n".format(
-        installed_s, latest_s
-    )
+def _format_core_msg(lines: List[List[str]]) -> str:
+    msg = "Core:\n"
+    msg_lines = []
 
-    plugin_version_msg = "Plugins:\n"
-    for plugin_name, version in _get_dbt_plugins_info():
-        plugin_version = dbt.semver.VersionSpecifier.from_version_string(version)
-        latest_plugin_version = get_latest_version(version_url=get_package_pypi_url(plugin_name))
-        plugin_update_msg = ""
-        if installed == plugin_version or (
-            latest_plugin_version and plugin_version == latest_plugin_version
-        ):
-            compatibility_msg = green("Up to date!")
-        else:
-            if latest_plugin_version:
-                if installed.major == plugin_version.major:
-                    compatibility_msg = yellow("Update available!")
-                else:
-                    compatibility_msg = red("Out of date!")
-                plugin_update_msg = (
-                    "  Your version of dbt-{} is out of date! "
-                    "You can find instructions for upgrading here:\n"
-                    "  https://docs.getdbt.com/dbt-cli/install/overview\n\n"
-                ).format(plugin_name)
-            else:
-                compatibility_msg = yellow("No PYPI version available")
-
-        plugin_version_msg += ("  - {}: {} - {}\n" "{}").format(
-            plugin_name, version, compatibility_msg, plugin_update_msg
-        )
+    for name, version, update_msg in _pad_lines(lines, seperator=":"):
+        line_msg = f"  - {name} {version}"
+        if update_msg != "":
+            line_msg += f" - {update_msg}"
+        msg_lines.append(line_msg)
 
-    if latest is None:
-        return (
-            "{}The latest version of dbt could not be determined!\n"
-            "Make sure that the following URL is accessible:\n{}\n\n{}".format(
-                version_msg, PYPI_VERSION_URL, plugin_version_msg
-            )
-        )
+    return msg + "\n".join(msg_lines)
+
+
+def _get_plugins_msg(installed: Type[dbt.semver.VersionSpecifier]) -> str:
+    msg = "Plugins:\n"
+    msg_lines = []
 
-    if installed == latest:
-        return f"{version_msg}{green('Up to date!')}\n\n{plugin_version_msg}"
+    plugins = []
+    plugin_msgs = []
+    for name, version_s in _get_dbt_plugins_info():
+        compatability_msg, update_msg = _get_plugin_msg_info(name, version_s, installed)
+        plugins.append([name, version_s, compatability_msg])
+        plugin_msgs.append(update_msg)
 
-    elif installed > latest:
-        return "{}Your version of dbt is ahead of the latest " "release!\n\n{}".format(
-            version_msg, plugin_version_msg
+    for i, plugin in enumerate(_pad_lines(plugins, seperator=":")):
+        msg_lines.append(_format_single_plugin(plugin, plugin_msgs[i]))
```

Comment:
[More significant feedback, shouldn't block initial merge]

I see that `_format_single_plugin` combines both the `msg` (short one-liners) and the `update_msg` (longer, more text). Was your thinking here that all lengthier "update" messages should actually go at the bottom, alongside the core update info?

To achieve that, I'd think about factoring the code like:
- `_format_single_plugin` accepts `plugin` and returns just that plugin's one-liner
- This method (`_get_plugins_msg`) returns each plugin's one-liner, and update messages (if required)
- `get_version_information()` combines all messages to create the following structure:
```
Core:
  - installed: ...
  - latest: ...

Plugins:
  - <one liner for each plugin>

<More text here if update message needed>
```

I gave that a quick go, but found the update messages get redundant quickly if you have multiple plugins installed:
```
$ dbt --version
Core:
  - installed: 1.0.2
  - latest:    1.0.4 - Update available!

Plugins:
  - postgres:  1.0.3   - Update available!
  - redshift:  1.1.0b1 - Not compatible!
  - snowflake: 1.1.0b1 - Not compatible!
  - bigquery:  1.1.0b1 - Not compatible!
  - spark:     1.1.0b1 - Not compatible!

Your version of dbt is out of date! You can find instructions for upgrading here:
https://docs.getdbt.com/docs/installation

  Your version of dbt-postgres is out of date! You can find instructions for upgrading here:
  https://docs.getdbt.com/dbt-cli/install/overview
  Your version of dbt-redshift is not compatible with core!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/dbt-cli/install/overview
  Your version of dbt-snowflake is not compatible with core!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/dbt-cli/install/overview
  Your version of dbt-bigquery is not compatible with core!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/dbt-cli/install/overview
  Your version of dbt-spark is not compatible with core!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/dbt-cli/install/overview
```

At which point, I'd almost rather just make it say:
```
$ dbt --version
Core:
  - installed: 1.0.2
  - latest:    1.0.4 - Update available!

Plugins:
  - postgres:  1.0.3   - Update available!
  - redshift:  1.1.0b1 - Not compatible!
  - snowflake: 1.1.0b1 - Not compatible!
  - bigquery:  1.1.0b1 - Not compatible!
  - spark:     1.1.0b1 - Not compatible!

Your installed version of dbt-core is out of date.
At least one plugin is out of date or incompatible with dbt-core.
You can find instructions for upgrading here:
https://docs.getdbt.com/docs/installation
```

Reply:
Could this be something like the following, so the user has a bit more context for which message belongs to which section?
```
$ dbt --version
Core:
  - installed: 1.0.2
  - latest:    1.0.4 - Update available!

  Your installed version of dbt is out of date.
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - postgres:  1.0.3   - Update available!
  - redshift:  1.1.0b1 - Not compatible!
  - snowflake: 1.1.0b1 - Not compatible!
  - bigquery:  1.1.0b1 - Not compatible!
  - spark:     1.1.0b1 - Not compatible!

  At least one plugin is out of date or incompatible with dbt.
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation
```

Reply:
I like tha!